UML_class.md

```diff
@@ -0,0 +1,334 @@
+```mermaid
```

Comment:
I assume this is just for documenting and it's not used by code anywhere, right?

Reply:
Yea, this was a commit I had at the end of last quarter that I have closed once I put up this one - basically this is just for the UML diagrams

Reply:
I can reopen that PR if you want to see those changes on their own

Reply:
That's ok. I just wanted to be sure I understood.